pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -163,6 +164,12 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env *model.Environme
 	for _, service := range push.Services(proxy) {
 		destRule := push.DestinationRule(proxy, service)
 		for _, port := range service.Ports {
+			destServiceMTLSMode := authn_v1alpha1_applier.MTLSUnknown
+			// For now, require enableAutoMtls mode set (to true) explicitly to recorgnize this mode.
+			if env.Mesh.EnableAutoMtls != nil && env.Mesh.EnableAutoMtls.GetValue() {
+				destServiceMTLSMode = authn_v1alpha1_applier.GetMutualTLSMode(
```

Comment:
and this check on env.Mesh.EnableAutoMtls is not necessary as this code applies in all cases.